### PR TITLE
Adds numerous golem types, some unused, and adds abilities to most

### PR DESCRIPTION
**Pull request by ghost**
ports a lot of tgs golems, adds a shit one of my own

1. preternis will now appear visibly ugly af when hit below a certain health threshhold/fixes some fluf messages

2. Golem types! Now free golems have variety.

Plasma golems explode on death, can be dismembered, and can burn.

Diamond golems are tough.

Gold golems are not tough, but faster than the norm.

Silver golems stun more often on hit.

Uranium golems constantly spam rad pulses. So no that doesn't kill golems but they will murder visitors accidentalyl.

Plasteel golems hit like a truck and stun often but are slow as hell.

Alien alloy golems are fast, can use guns, repair over time, but are silent and can only speak with other alloy golems.

Sand golems are immune to bullets but weak as hell against burn and have no natural armor.

Wood golems are weak to burn but heal in the light like podmen.

Mythril golems are 100% immune to magical projectiles, hit like a truck, don't stun, when stunned stay down three times as long, and take ridiculous damage from being shocked. They also move slow as fuck but are basically immune to god knows everything-they'll take a hell of a beating.

3. Plasmamen are now not virusimmune and can also dismember and attatch easily since they're basically fire skeletons
#### Changelog

:cl:
rscadd: Different types of golems have different innate abilities. Shamelessly stolen from /tg/.

/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
this is the PR https://github.com/tgstation/tgstation/pull/21708

also I'm pretty sure that plasmamen still won't transfer viruses since their suits are thick

**Comment by ShadowDeath6** [comment]
also I THINK that golems don't have limb sprites so they have to be nodismember

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 1]
```diff
@@ -820,27 +823,181 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
+	fixed_mut_color = "ffa07a"
+	armor = 75 //tough af
+	burnmod = 0 //immune to burn entirely
+	brutemod = 0.3 //tough af against brute
+	heatmod = 0 //immune to heat
+	siemens_coefficient = 2 //don't shock them tho
+	speedmod = 6 //slow as FUCK
+	stunmod = 3 //if they go down they stay down
+	punchdamagelow = 9 //can either punch low...
+	punchdamagehigh = 25 //or really hard
```
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/species_types.dm:976]: make them closer, nobody is going to want to play as a cardboard box that can do half as much or 4 more damage than a regular plasteel golem

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 2]
```diff
@@ -820,27 +823,181 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
+	punchstunthreshold = 50 //but it'll never stun
+	
+	
+/datum/species/golem/mythril/spec_death(gibbed, mob/living/carbon/human/H)
+	H.visible_message("<span class='danger'>[H] turns into dust!</span>")
+	for(var/obj/item/W in H)
+		H.unEquip(W)
+	for(var/i=1, i <= rand(3,5), i++)
+		new /obj/item/weapon/ore/glass(get_turf(H))
```
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/species_types.dm:985]: why glass?

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 3]
```diff
@@ -820,27 +823,181 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
+	qdel(H)
+	
+/datum/species/golem/proc/on_hit(obj/item/projectile/proj_type, mob/living/carbon/human/H)
+	switch(proj_type)
+		if(/obj/item/projectile/magic)
+			playsound(H, "sound/effects/shovel_dig.ogg", 70, 1)
```
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/species_types.dm]: no magic noise?

**Comment by ShadowDeath6** [comment]
tl;dr mythril golem will do jack shit because 1. golems can't beat ashwalkers and 2. gets stunned by goliath tendril for 8 seconds because you can't dodge and get hit like 15 times until ded

**Comment by Dagdammit** [comment]
"Limbs reattach easily": How so? Just use the limb on the stump, with maybe a 10+ second action to complete?

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 2 above)
Comment (XDTM) [reply code/modules/mob/living/carbon/human/species_types.dm:985]: That's sand

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 4]
```diff
@@ -186,7 +186,9 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
 	burnmod = 1.05
 	heatmod = 1.05
 	invis_sight = SEE_INVISIBLE_MINIMUM
+	disguise_fail_health = 75
```
Comment (Time-Green) [inline code/modules/mob/living/carbon/human/species_types.dm]: Adam, this was defined at \synth
And since preternisses aren't synth subtypes, the var doesn't exist here.
Also it's a bad change anyway

**Comment by Cruix** [comment]
Golems have limb sprites since #1167

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 3 above)
Comment (Cruix) [reply code/modules/mob/living/carbon/human/species_types.dm]: This has to be an istype(), the switch will only catch things that are <i>exactly</i> that type, not subtypes.

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 1 above)
Comment (Cruix) [reply code/modules/mob/living/carbon/human/species_types.dm:976]: It's fine, they are almost immune to brute and burn damage, they don't need to deal tons of damage unarmed as well.

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 5]
```diff
@@ -820,27 +823,181 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
+
+/datum/species/golem/sand/spec_death(gibbed, mob/living/carbon/human/H)
+	H.visible_message("<span class='danger'>[H] turns into a pile of sand!</span>")
+	for(var/obj/item/W in H)
+		H.unEquip(W)
+	for(var/i=1, i <= rand(3,5), i++)
+		new /obj/item/weapon/ore/glass(get_turf(H))
+	qdel(H)
+
+/datum/species/golem/proc/on_hit(obj/item/projectile/proj_type, mob/living/carbon/human/H)
```
Comment (Cruix) [inline code/modules/mob/living/carbon/human/species_types.dm]: Change /proc/ to /sand/

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 6]
```diff
@@ -820,27 +823,181 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
+	switch(proj_type)
+		if(/obj/item/projectile/bullet) || (/obj/item/projectile/kinetic)
```
Comment (Cruix) [inline code/modules/mob/living/carbon/human/species_types.dm]: You can't use swich for complicated stuff like this, just do an if with two istype()s

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 4 above)
Comment (Cruix) [reply code/modules/mob/living/carbon/human/species_types.dm]: I'm fairly certain you can change the preternis'a damage overlays to the same as synths, disguise_fail_health is only used for little touches like spec_life() and handle_speech() now.

**Comment by Cruix** [comment]
This is very, very good.

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 4 above)
Comment (Time-Green) [reply code/modules/mob/living/carbon/human/species_types.dm]: I know you can do it, but I don't like it. What I pointed out is also an error

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 1 above)
Comment (ShadowDeath6) [reply code/modules/mob/living/carbon/human/species_types.dm:976]: all people are going to do is stay back 10 tiles and tase them and hit them to death

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 2 above)
Comment (ShadowDeath6) [reply code/modules/mob/living/carbon/human/species_types.dm:985]: this is mythril golems dropping sand on death

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 6 above)
Comment (XDTM) [reply code/modules/mob/living/carbon/human/species_types.dm]: if you use on_hit it'll proc after the bullet deals damage, you need to make a species bullet_act

**Comment by Super3222** [comment]
Reference /tg/station, and the /tg/station developer in your changelog, please.

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/species_types.dm:985]: well i don't want them dropping mythril since i have plans for it

ideally mythril would be consumed after use blah blah magic

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/species_types.dm:976]: these guys are basically immortal with 70 armor AND only taking a fraction of incoming damage, stunning them is the only way to kill them or they'll just keep advancing

**Comment by ghost** [comment]
i would make a species bullet_act but when I tried to do that shit went haywire so I might try to do that separately or go about it roundabout like having the proc treat the damage of the projectile immediately after it does it's effects but that could cause issues

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 2 above)
Comment (ShadowDeath6) [reply code/modules/mob/living/carbon/human/species_types.dm:985]: why are they dropping sand on death, just have them qdel or something

**Comment by XDTM** [comment]
I did that for my pr in /tg/ if you want to check it out, the relevant changes are in human_defense.dm and species.dm.

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 1 above)
Comment (XDTM) [reply code/modules/mob/living/carbon/human/species_types.dm:976]: The slowness is enough to keep the distance, triple stuntimes means that you can taze a mythril golem, pop the taser in a charger, go have lunch, come back and do it all over again
